src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -112,6 +119,19 @@ const initialState: InitialState = {
   deviceState: initialDeviceState,
 }
 
+export const isKeyManagerWithProvider = (
+  keyManager: KeyManager | null,
```

Comment:
Not sure if this is more proper:

```suggestion
  keyManager: KeyManager | KeyManagerWithProvider | null,
```

Reply:
Also valid point! The way I see type guards, it goes from broad in the arguments, to narrowed in the return type, and `KeyManagerWithProvider` is just a subset of `KeyManager` so IMO it doesn't make sense to have `Broad | Narrowed` as input type

Reply:
I buy that 🛍️